airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -2564,7 +2564,9 @@ def create_record_selector(
             else None
         )
 
-        transform_before_filtering = False
+        assert model.transform_before_filtering is not None  # for mypy
```

Comment:
Can we avoid using `assert` here, it's great we can use it, but if it's just to satisfy the `MyPy` checker - let's use `# type: ignore` instead. WDYT?

Reply:
Yes, we have same parts throughout this file. I think we need to replace them all at once. For now, let’s leave it as is.